src/components/elements/pagination.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { navigate } from 'gatsby'
+import { Button } from 'components/form'
+import { Flex } from 'components/containers'
+import { queryParams } from 'common/utility'
+
+type PaginationProps = {
+    items_per_page: number
+    total_items: number
+    paginate: (arg1: number) => void
+    current_page: number
+}
+
+const StyledFlex = styled(Flex)`
+    margin-bottom: 40px;
+    align-items: center;
+`
+
+const StyledButton = styled(Button)<{ active: boolean; grey: boolean }>`
+    border: 1px solid var(--color-grey-2);
+    padding: 7px 13px;
+    font-size: 14px;
+    margin: 0 4px;
+    color: ${(props) => (props.grey ? 'var(--color-grey-2)' : 'var(--color-black-3)')};
+    font-weight: normal;
+    background: ${(props) => props.disabled && 'var(--color-grey-2)'};
+    ${(props) => {
+        if (props.active)
+            return css`
+                pointer-events: none;
+                border: 1px solid var(--color-black-3);
+            `
+    }};
+`
+
+const Pagination = ({ items_per_page, total_items, paginate, current_page }: PaginationProps) => {
+    const page_numbers = []
+
+    const total_pages = Math.ceil(total_items / items_per_page)
+
+    for (let i = 1; i <= total_pages; i++) {
+        page_numbers.push(i)
+    }
+
+    const last_page = page_numbers.length
+
+    const sliced_page_numbers =
+        last_page > 5
+            ? page_numbers.slice(current_page > 2 && current_page - 1, current_page + 1)
```

Comment:
Can we assign floating numbers to  a meaningful variables, `5` and `2` are not clear to us what is  it referring to.